articles/agora-cloud-recording/index.md

```diff
@@ -0,0 +1,496 @@
+
+### Recording
+We need to use the RESTful APIs in the following sequence.
+
+1. Acquire resource.
+2. Start recording.
+3. Stop recording.
+
+First, you need to acquire a resource ID for cloud recording. This will allocate a recorder on Agora's servers for you to record the streams. Then, we need to start the recording within 5 minutes from acquiring the resource ID. You can stop the recording whenever you want.
+
+During the recording, you can query the recording session for the status, update the layout of the recording, and, update the subscriber list. To learn more about them, refer to [the documentation](https://docs.agora.io/en/cloud-recording/cloud_recording_rest?platform=RESTful).
+
+We need to provide a UID for the recorder. The recorder is like a user who joins the channel, records the stream, and uploads it to the storage. So make sure you provide a unique UID to the recorder that doesn't conflict with an existing user in the channel.
+
+If you have set up token authentication for your Agora project, you should also pass a `token` in the body. To learn more about token authentication, refer to [this article](engineering-education/agora-express-token-server/).
+
+> At the time of writing this article, Agora cloud recording does not support user accounts yet. Make sure that the recording channel uses integer UIDs. If you'd like to learn more about user accounts, refer to [this FAQ answer](https://docs.agora.io/en/All/faq/string).
+
+### Acquire resource ID
+Let's add a POST request handler for a new endpoint called `'/acquire'` to acquire the resource ID for the cloud recording.
+
+```JavaScript
+app.post("/acquire", (req, res) => {
+  // Acquire Cloud Recording Resource ID Here
+});
+```
+
+You need to request this endpoint `https://api.agora.io/v1/apps/{appId}/cloud_recording/acquire` to get a resource ID. The endpoint URL should contain the Agora app ID.
+
+In the body of the request, We should specify a UID, the channel ID, and the time limit (in hours) for the cloud recording API calls. The time limit must be between 1 hour and 720 hours. The countdown starts when you start the recording. When you exceed the limit, you can no longer call `query`, `updateLayout`, and `stop`.
+
+If this request succeeds, you'll get a resource ID as the response. You need to start recording with this resource ID within five minutes.
+
+```JavaScript
+app.post("/acquire", async (req, res) => {
+  const Authorization = `Basic ${Buffer.from(`${process.env.RESTkey}:${process.env.RESTsecret}`).toString("base64")}`;
+  const acquire = await axios.post(
+    `https://api.agora.io/v1/apps/${process.env.appID}/cloud_recording/acquire`,
+    {
+      cname: req.body.channel,
+      uid: req.body.uid,
+      clientRequest: {
+        resourceExpiredHour: 24,
+      },
+    },
+    { headers: { Authorization } }
+  );
+
+  res.send(acquire.data);
+});
+```
```

Comment:
I got stuck here
1. You have not added Axios as a requirement
2. The async request I make throws an error. And since there's no error handling there's no clear way to know what's going wrong.
3. You did not include guidance on how to go about obtaining the APP_ID variable (which I'm assuming is why the code isn't working, I'm not certain)

Reply:
![image](https://user-images.githubusercontent.com/45763204/103767947-905f5900-5047-11eb-8f30-d1a5632df14e.png)

I did add Axios as a requirement. I'll update the article with an `npm install` command to make it stand out.

Reply:
If you didn't include an APP_ID, the requests will definitely throw an error.

Reply:
![acquire](https://user-images.githubusercontent.com/45763204/103770737-7ffdad00-504c-11eb-95ba-6fc07857f752.gif)